workos/resources/webhooks.py

```diff
@@ -0,0 +1,283 @@
+from typing import Generic, Literal, Union
+from pydantic import Field
+from typing_extensions import Annotated
+from workos.resources.directory_sync import DirectoryGroup
+from workos.resources.events import EventPayload
+from workos.resources.user_management import OrganizationMembership, User
+from workos.resources.workos_model import WorkOSModel
+from workos.types.directory_sync.directory_user import DirectoryUser
+from workos.types.events.authentication_payload import (
+    AuthenticationEmailVerificationSucceededPayload,
+    AuthenticationMagicAuthFailedPayload,
+    AuthenticationMagicAuthSucceededPayload,
+    AuthenticationMfaSucceededPayload,
+    AuthenticationOauthSucceededPayload,
+    AuthenticationPasswordFailedPayload,
+    AuthenticationPasswordSucceededPayload,
+    AuthenticationSsoSucceededPayload,
+)
+from workos.types.events.connection_payload_with_legacy_fields import (
+    ConnectionPayloadWithLegacyFields,
+)
+from workos.types.events.directory_group_membership_payload import (
+    DirectoryGroupMembershipPayload,
+)
+from workos.types.events.directory_group_with_previous_attributes import (
+    DirectoryGroupWithPreviousAttributes,
+)
+from workos.types.events.directory_payload import DirectoryPayload
+from workos.types.events.directory_payload_with_legacy_fields import (
+    DirectoryPayloadWithLegacyFields,
+)
+from workos.types.events.directory_user_with_previous_attributes import (
+    DirectoryUserWithPreviousAttributes,
+)
+from workos.types.events.organization_domain_verification_failed_payload import (
+    OrganizationDomainVerificationFailedPayload,
+)
+from workos.types.events.session_created_payload import SessionCreatedPayload
+from workos.types.organizations.organization_common import OrganizationCommon
+from workos.types.organizations.organization_domain import OrganizationDomain
+from workos.types.roles.role import Role
+from workos.types.sso.connection import Connection
+from workos.types.user_management.email_verification_common import (
+    EmailVerificationCommon,
+)
+from workos.types.user_management.invitation_common import InvitationCommon
+from workos.types.user_management.magic_auth_common import MagicAuthCommon
+from workos.types.user_management.password_reset_common import PasswordResetCommon
+
+
+class WebhookModel(WorkOSModel, Generic[EventPayload]):
+    """Representation of an Webhook delivered via Webhook.
+    Attributes:
+        OBJECT_FIELDS (list): List of fields an Webhook is comprised of.
```

Comment:
This comment can go since `OBJECT_FIELDS` doesn't exist anymore.

Reply:
Yeah, I plan to do a comment/docstring sweep across all the resources as a follow-up. Though, I haven't seen any place where these docstrings are surfaced to the developer. Do you know how they're usually used?

Reply:
I see them in my editor, albeit poorly formatted:
![image](https://github.com/user-attachments/assets/52107c35-c638-4d19-8937-eb6a35c2beef)

Reply:
Interesting. It seems like they're more useful the the actual SDK methods. Not to much for the resources/models. I have to dig into the packages to actually see the docstrings for our resources.
![CleanShot 2024-07-31 at 10 52 18](https://github.com/user-attachments/assets/52b0fde5-8aed-45e7-aac5-aa89ba3c836a)